### PR TITLE
OJ-2556: Added Alarm for FailedInvocation Metric to EventBridge Rules (PART 1)

### PR DESCRIPTION
**Pull request by jkunle**
## Proposed changes 

A meaningful monitoring and alarms for the EventBridge rules

### What changed

Added FailedInvocation Alarm for Audit Event Eventbridge Rules in Nino CRI, Eventbridge rules have many alarms but this seems like a good place to start. This PR doesn't consider what happens to messages when things fails. 

NB: The way events are made available to the TxMA queue has been refactored in this PR. This now uses a new Event bridge rule `TxMaAuditEventRule` , which needs a FailedInvocation Alarm also included in this PR

See: https://gds.slack.com/archives/C06TX3H2XPS/p1715241837577809

- [OJ-2556](https://govukverify.atlassian.net/browse/OJ-2556)

[OJ-2556]: https://govukverify.atlassian.net/browse/OJ-2556?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: infrastructure/template.yaml [hunk 1]
```diff
@@ -1788,6 +1813,29 @@ Resources:
 
   ###############################################################################
 
+  BadEventBridgeStateMachineExecutionRole:
+    Type: AWS::IAM::Role
```
Comment (surajkumar) [inline infrastructure/template.yaml:1817]: Can you add a dev-like condition to this please?

File: step-functions/audit_event.asl.json [hunk 2]
```diff
@@ -191,12 +191,19 @@
 		},
 		"publish event to TxMa Queue": {
 			"Type": "Task",
-      "InputPath": "$[0]",
-			"Resource": "arn:aws:states:::sqs:sendMessage",
+      "InputPath": "$[0].auditEvent",
+			"Resource": "arn:aws:states:::events:putEvents",
```
Comment (surajkumar) [inline step-functions/audit_event.asl.json:195]: Can you explain what behaviour change this makes?

**Review by surajkumar — CHANGES_REQUESTED** [review]

File: step-functions/audit_event.asl.json (hunk 2 above)
Comment (jkunle) [reply step-functions/audit_event.asl.json:195]:
```						
"Detail.$": "$[0].auditEvent",
"DetailType.$": "$[0].auditEvent.event_name",
```

File: infrastructure/template.yaml (hunk 1 above)
Comment (jkunle) [reply infrastructure/template.yaml]: I have removed `BadEventBridgeStateMachineExecutionRole` it done it's job

**Review by surajkumar — DISMISSED** [review]

**Review by surajkumar — APPROVED** [review]

**Pull request merged**